### PR TITLE
[wms] Allow for WMTS layers to take DPI into account and scale up on high DPI screens

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR addresses a shortcoming of the WMS provider's handling of WMTS on high DPI screens by allowing scaling up of tiles being rendered based on the device screen DPI. A new combobox has been added in the new WMS/WMTS server dialog to setup the WMTS tile pixel ratio to handle scale up/down based on the rendered DPI:
image

_Tip: enabling smooth pixmap transform is a must when this auto scale up is active_

### CODE REVIEW

File: src/core/qgsowsconnection.cpp [hunk 1]
```diff
@@ -132,6 +132,11 @@ QgsDataSourceUri &QgsOwsConnection::addWmsWcsConnectionSettings( QgsDataSourceUr
   {
     uri.setParam( QStringLiteral( "dpiMode" ), dpiMode );
   }
+  const QString tilePixelRation = settings.value( settingsKey + QStringLiteral( "/tilePixelRation" ), "0" ).toString();
+  if ( tilePixelRation != QStringLiteral( "0" ) )
+  {
+    uri.setParam( QStringLiteral( "tilePixelRation" ), tilePixelRation );
```
Comment (nyalldawson) [inline src/core/qgsowsconnection.cpp:138]:
```suggestion
    uri.setParam( QStringLiteral( "tilePixelRatio" ), tilePixelRatio );
```

Comment (nirvn) [reply src/core/qgsowsconnection.cpp]: @nyalldawson , thanks, fixed.

File: src/gui/qgsnewhttpconnection.cpp [hunk 2]
```diff
@@ -65,13 +65,17 @@ QgsNewHttpConnection::QgsNewHttpConnection( QWidget *parent, ConnectionTypes typ
...
   cmbDpiMode->clear();
-
   cmbDpiMode->addItem( tr( "all" ), static_cast<int>( Qgis::DpiMode::All ) );
   cmbDpiMode->addItem( tr( "off" ), static_cast<int>( Qgis::DpiMode::Off ) );
   cmbDpiMode->addItem( tr( "QGIS" ), static_cast<int>( Qgis::DpiMode::QGIS ) );
   cmbDpiMode->addItem( tr( "UMN" ), static_cast<int>( Qgis::DpiMode::UMN ) );
   cmbDpiMode->addItem( tr( "GeoServer" ), static_cast<int>( Qgis::DpiMode::GeoServer ) );
 
+  cmbTilePixelRatio->clear();
+  cmbTilePixelRatio->addItem( tr( "Unknown (not scaled)" ), static_cast<int>( Qgis::TilePixelRatio::Unknown ) );
```
Comment (m-kuhn) [inline src/gui/qgsnewhttpconnection.cpp:75]: Can we call it "undefined" or "native" or "unscaled"?

Comment (nirvn) [reply src/gui/qgsnewhttpconnection.cpp]: @m-kuhn , picked undefined.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**